app/cluster/dynamic.go

```diff
@@ -125,6 +163,15 @@ func (d *Dynamic) stop() {
 	d.serverStopCountStat.Increment()
 }
 
+func (d *Dynamic) handleWorkspaceChange(workspaces string) error {
+	if d.currentWorkspaceIDs == workspaces {
+		return nil
+	}
+	d.BackendConfig.StopPolling()
```

Comment:
qq: is `StopPolling()` idempotent? What if we stop but polling was never started?

Reply:
`StopPolling()` is not idempotent. The assumption is that BackendConfig will always be running before dynamicCluster is started. 

I am concerned about this as well, I have some plans to refactor backend config to improve on this. This PR is already too big, I would like to do it in a separate one.